.github/workflows/deploy-frontend.yml

```diff
@@ -4,28 +4,37 @@ on:
   workflow_call:
     inputs:
       report_artifact_in_scope:
-        required: false
         type: boolean
+        default: false
   workflow_dispatch:
   release:
     types: [published]
 
 jobs:
+  # if called as a separate workflow
   build-frontend:
     uses: ./.github/workflows/build-frontend.yml
+    if: inputs.report_artifact_in_scope == false
+
   collect:
     runs-on: ubuntu-latest
+    if: ${{ !failure() }}
     needs: build-frontend
     steps:
       - name: Create _site directory
         run: mkdir _site
-      - name: Download frontend build
-        uses: actions/download-artifact@v4
+
+      # if called from report workflow
+      - name: Download latest frontend-build
+        uses: dawidd6/action-download-artifact@v3
+        if: inputs.report_artifact_in_scope == true
         with:
+          workflow: build-frontend.yml
+          branch: main
```

Comment:
@Julian the site seems to be functioning for now and I've downloaded the `github-pages` artifact that was generated. However I observed one thing that my fix of removing IMAGE_PREFIX from frontend did not show up on the website as in the above code downloading the latest frontend-build from `main` branch of `build-frontend.yml` did not have that fix of mine. So what do you think should be the best place to get the last / latest frontend-build from ?

Reply:
I think this might also be the cause of previously failed efforts

Reply:
It should probably always use the current branch for the workflow (which I'm guessing but haven't checked should be the default if you don't specify branch) but then we should never deploy from anywhere other than main

Reply:
Yeah I checked if you dont specify any branch explicitly then it uses the current branch. So do you mean i should remove `branch: main` from here ?

Reply:
Yes just double check that somewhere below only runs if it's main

Reply:
Meaning ?

Reply:
What I mentioned above -- it's fine to build every branch using its own workflow, but we should be sure to only finish / deploy if we're running on main.

Reply:
Okay got it!

Reply:
> It should probably always use the current branch for the workflow (which I'm guessing but haven't checked should be the default if you don't specify branch) but then we should never deploy from anywhere other than main

but I am guessing the problem here is that the current branch won't necessarily have ran the build-frontend workflow since it is purely relying on it to just get the artifacts that have already been ran from some other branch right ?

Reply:
Sorry nevermind I figured it out :) Thanks!